client/hotrod-client/src/test/java/org/infinispan/client/hotrod/counter/StrongCounterAPITest.java

```diff
@@ -0,0 +1,117 @@
+package org.infinispan.client.hotrod.counter;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import org.infinispan.counter.api.CounterListener;
+import org.infinispan.counter.api.Handle;
+import org.infinispan.counter.api.StrongCounter;
+import org.infinispan.server.hotrod.counter.StrongCounterTest;
+import org.infinispan.server.hotrod.counter.impl.StrongCounterImplTest;
+import org.testng.annotations.Test;
+
+/**
+ * A {@link StrongCounter} implementation test.
+ *
+ * @author Pedro Ruivo
+ * @since 9.2
+ */
+@Test(groups = "functional", testName = "client.hotrod.counter.StrongCounterAPITest")
+public class StrongCounterAPITest extends BaseCounterAPITest<StrongCounter> implements StrongCounterTest {
+
+   private final StrongCounterImplTest testImpl;
+
+   public StrongCounterAPITest() {
+      testImpl = new StrongCounterImplTest(this::counterManager, this::counterManagers);
+   }
+
+   @Override
+   public void testCompareAndSet(Method method) {
+      testImpl.testCompareAndSet(method);
+   }
+
+   @Override
+   public void testCompareAndSwap(Method method) {
+      testImpl.testCompareAndSwap(method);
+   }
+
+   @Override
+   public void testBoundaries(Method method) {
+      testImpl.testBoundaries(method);
+   }
+
+   @Override
+   public void testListenerWithBounds(Method method) throws InterruptedException {
+      testImpl.testListenerWithBounds(method);
+   }
+
+   @Override
+   public void testAdd(Method method) {
+      testImpl.testAdd(method);
+   }
+
+   @Override
+   public void testReset(Method method) {
+      testImpl.testReset(method);
+   }
+
+   @Override
+   public void testNameAndConfigurationTest(Method method) {
+      testImpl.testNameAndConfigurationTest(method);
+   }
+
+   @Override
+   public void testRemove(Method method) {
+      testImpl.testRemove(method);
+   }
+
+   @Override
+   public void testListenerAddAndRemove(Method method) throws InterruptedException {
+      testImpl.testListenerAddAndRemove(method);
+   }
+
+   @Override
+   public void testExceptionInListener(Method method) throws InterruptedException {
+      super.testExceptionInListener(method);
```

Comment:
why do you need to override this?

Reply:
the `StrongCounterTest` is an interface and this way I make sure that I'll run all the tests :)

Reply:
I thought that this is what the interface enforces...

Reply:
nevermind, I was looking at another test :) 
I didn't mark the base class as test and I need to override here.

Reply:
IIRC you could put the `@Test` annotation on the super method.